ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/WorkloadCvesImageSinglePage.tsx

```diff
@@ -1,9 +1,205 @@
+import ImageSingleResources from './ImageSingleResources';
+import useDetailsTabParameter from './hooks/useDetailsTabParameter';
+import { isDetailsTab } from './types';
+
+const workloadCveOverviewImagePath = `${vulnerabilitiesWorkloadCvesPath}?cveStatusTab=Observed&entityTab=Image`;
+
+function ImageDetailBadges({ imageData }: { imageData: ImageDetailsResponse['image'] }) {
+    const [hasSuccessfulCopy, setHasSuccessfulCopy] = useState(false);
+
+    const { deploymentCount, operatingSystem, metadata, scan } = imageData;
+    const created = metadata?.v1?.created;
+    const sha = metadata?.v1?.digest;
+    const isActive = deploymentCount > 0;
+
+    function copyToClipboard(imageSha: string) {
+        navigator.clipboard
+            .writeText(imageSha)
+            .then(() => setHasSuccessfulCopy(true))
+            .catch(() => {
+                // Permission is not required to write to the clipboard in secure contexts when initiated
+                // via a user event so this Promise should not reject
+            })
+            .finally(() => {
+                setTimeout(() => setHasSuccessfulCopy(false), 2000);
+            });
+    }
+
+    return (
+        <LabelGroup numLabels={Infinity}>
+            <Label isCompact color={isActive ? 'green' : 'gold'}>
+                {isActive ? 'Active' : 'Inactive'}
+            </Label>
+            {operatingSystem && <Label isCompact>OS: {operatingSystem}</Label>}
+            {created && (
+                <Label isCompact>Age: {getDistanceStrictAsPhrase(created, new Date())}</Label>
+            )}
+            {scan && (
+                <Label isCompact>
+                    Scan time: {getDateTime(scan.scanTime)} by {scan.dataSource.name}
+                </Label>
+            )}
+            {sha && (
+                <Tooltip content="Copy image SHA to clipboard">
+                    <Label
+                        style={{ cursor: 'pointer' }}
+                        icon={<CopyIcon />}
+                        isCompact
+                        color={hasSuccessfulCopy ? 'green' : 'grey'}
+                        onClick={() => copyToClipboard(sha)}
+                    >
+                        {hasSuccessfulCopy ? 'Copied!' : 'SHA'}
+                    </Label>
+                </Tooltip>
+            )}
+        </LabelGroup>
+    );
+}
+
 function WorkloadCvesImageSinglePage() {
     const { imageId } = useParams();
-    return <>Workload CVE Image Single Page: {imageId}</>;
+    const { data, error } = useQuery<ImageDetailsResponse, ImageDetailsVariables>(
+        imageDetailsQuery,
+        {
+            variables: { id: imageId },
+        }
+    );
+
+    const [activeTabKey, setActiveTabKey] = useDetailsTabParameter();
+
+    const imageData = data && data.image;
+    const imageName = imageData?.name?.fullName ?? 'NAME UNKNOWN';
+
+    const handleTabClick: TabsProps['onSelect'] = (e, tabKey) => {
```

Comment:
oh neat i wasn't aware we could do that to selectively pull props from the type defs of the library

Reply:
This was super useful during collections to avoid having to duplicate type signatures. Especially in situations where there was prop-drilling but it didn't really make sense to use a context; we could define the prop type of a component in terms of the prop type of a child component, and when things changed we only needed to change it in two places instead of in every intermediate component.